src/frontend/src/pages/FinancePage/ReimbursementRequestsSection.tsx

```diff
@@ -26,13 +34,79 @@ const ReimbursementRequestTable = ({
   const displayedReimbursementRequests =
     tabValue === 1 && allReimbursementRequests ? allReimbursementRequests : userReimbursementRequests;
 
-  const rows = displayedReimbursementRequests
-    .map(createReimbursementRequestRowData)
-    .sort((a, b) => (a.dateSubmitted > b.dateSubmitted ? -1 : 1));
+  // function getComparator<Key extends keyof any>(
+  //   order: Order,
+  //   orderBy: Key
+  // ): (a: { [key in Key]: number | string }, b: { [key in Key]: number | string }) => number {
+  //   return order === 'desc' ? (a, b) => descendingComparator(a, b, orderBy) : (a, b) => -descendingComparator(a, b, orderBy);
+  // }
+
+  function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
+    if (b[orderBy] < a[orderBy]) {
+      return -1;
+    }
+    if (b[orderBy] > a[orderBy]) {
+      return 1;
+    }
+    return 0;
+  }
+
+  function statusDescendingComparator(a: ReimbursementStatusType, b: ReimbursementStatusType): number {
+    const statusOrder = new Map<ReimbursementStatusType, number>([
+      [ReimbursementStatusType.PENDING_FINANCE, 1],
+      [ReimbursementStatusType.SABO_SUBMITTED, 2],
+      [ReimbursementStatusType.ADVISOR_APPROVED, 3],
+      [ReimbursementStatusType.DENIED, 5]
+    ]);
+
+    let bConverted = statusOrder.get(b);
+    let aConverted = statusOrder.get(a);
+
+    if (bConverted !== undefined && aConverted !== undefined) {
+      if (bConverted < aConverted) {
+        return -1;
+      }
+      if (bConverted > aConverted) {
+        return 1;
+      }
+    }
+    return 0;
+  }
+
+  function vendorDescendingComparator(a: Vendor, b: Vendor): number {
```

Comment:
Let's use an `arrow function` instead of `function` for the consistency in our codebase